File: lib/cleverpush_chat_view.dart
import 'package:flutter/foundation.dart';
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
import 'package:flutter/services.dart';

class CleverPushChatView extends StatelessWidget {
  const CleverPushChatView({ Key? key }) : super(key: key);

  Widget build(BuildContext context) {
    const String viewType = 'cleverpush-chat-view';
    const Map<String, dynamic> creationParams = <String, dynamic>{};

    switch (defaultTargetPlatform) {
        case TargetPlatform.android:
          return PlatformViewLink(
            viewType: viewType,
            surfaceFactory:
                (context, controller) {
              return AndroidViewSurface(
                controller: controller as AndroidViewController,
                gestureRecognizers: const <Factory<OneSequenceGestureRecognizer>>{},
                hitTestBehavior: PlatformViewHitTestBehavior.opaque,
              );
            },
            onCreatePlatformView: (params) {
              return PlatformViewsService.initSurfaceAndroidView(
                id: params.id,
                viewType: viewType,
                layoutDirection: TextDirection.ltr,
                creationParams: creationParams,
                creationParamsCodec: const StandardMessageCodec(),
                onFocus: () {
                  params.onFocusChanged(true);
                },
              )
                ..addOnPlatformViewCreatedListener(params.onPlatformViewCreated)
                ..create();
            },
          );
        case TargetPlatform.iOS:
          return UiKitView(
            viewType: viewType,
            layoutDirection: TextDirection.ltr,
            creationParams: creationParams,
            creationParamsCodec: const StandardMessageCodec(),
          );
        default:
          throw UnsupportedError('Unsupported platform view');
      }
  }
}

File: android/src/main/java/com/cleverpush/flutter/CleverPushChatView.java
package com.cleverpush.flutter;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.cleverpush.chat.ChatView;

import java.util.Map;

import io.flutter.plugin.platform.PlatformView;

class CleverPushChatView implements PlatformView {
    @NonNull private final ChatView chatView;

    CleverPushChatView(@NonNull Context context, int id, @Nullable Map<String, Object> creationParams) {
        chatView = new ChatView(context);
    }

    @NonNull
    @Override
    public View getView() {
        return chatView;
    }

    @Override
    public void dispose() {}
}
